src/devices/machine/acorn_vidc.h

```diff
@@ -0,0 +1,145 @@
+// license:LGPL-2.1+
+// copyright-holders:Angelo Salese, R. Belmont, Juergen Buchmueller
+/***************************************************************************
+
+	Acorn VIDC10 (VIDeo Controller) device chip
+
+***************************************************************************/
+
+#ifndef MAME_MACHINE_ACORN_VIDC_H
+#define MAME_MACHINE_ACORN_VIDC_H
+
+#pragma once
+
+#include "screen.h"
+#include "speaker.h"
+#include "emupal.h"
+#include "sound/dac.h"
+#include "sound/volt_reg.h"
+
+//**************************************************************************
+//  INTERFACE CONFIGURATION MACROS
+//**************************************************************************
+
+
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> acorn_vidc10_device
+
+class acorn_vidc10_device : public device_t,
+							public device_memory_interface,
+							public device_video_interface
+{
+public:
+	// construction/destruction
+	acorn_vidc10_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	// I/O operations
+	DECLARE_WRITE32_MEMBER( write );
+	DECLARE_READ_LINE_MEMBER( flyback_r );
+	auto vblank() { return m_vblank_cb.bind(); }
+	auto sound_drq() { return m_sound_drq_cb.bind(); }
+	// MEMC comms
+	void write_vram(uint32_t offset, uint8_t data) { m_data_vram[offset & (m_data_vram_mask)] = data; }
+	void write_cram(uint32_t offset, uint8_t data) { m_cursor_vram[offset & (m_cursor_vram_mask)] = data; }
+	void write_dac(uint8_t channel, uint8_t data);
+	void clear_dac(uint8_t channel) { m_dac[channel & 7]->write(0); }
+	void update_sound_mode(bool state) { m_sound_mode = state; refresh_sound_frequency(); }
+	void set_cursor_enable(bool state) { m_cursor_enable = state; }
+	uint32_t get_cursor_size() { return (m_crtc_regs[CRTC_VCER] - m_crtc_regs[CRTC_VCSR]) * (32/4); }
+	
+protected:
+	acorn_vidc10_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, u32 clock);
+
+	// device-level overrides
+	//virtual void device_validity_check(validity_checker &valid) const override;
+	virtual void device_add_mconfig(machine_config &config) override;
+	virtual void device_start() override;
+	virtual void device_reset() override;
+	virtual void device_timer(emu_timer &timer, device_timer_id id, int param, void *ptr) override;
+	virtual space_config_vector memory_space_config() const override;
+	u32 screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect);
+
+	required_device<screen_device> m_screen;
```

Comment:
Why does it have a `m_screen` shadowing the member with the same name in `device_video_interface`?

Reply:
Wasn't aware there's a m_screen in device_video_interface. In case I want to subclass that part how do I even then?

Reply:
Why do you need a special kind of `screen_device`?  What does it actually need to do?

Reply:
VIDC is special in how it handles vblank, want something that actually breaks on vblank signal with the debugger. Additionally if it has internal timers that will help with hsync DMA fetches, especially on performance (which is already at limit as things stands).

Reply:
The thing is, `device_video_interface` barely even gets you anything.  It finds a screen, if you don't specify one explicitly it looks for the first screen in the system, and it waits for the screen to start.  But none of that's hard to do anyway.  Do you really need to derive from `device_video_interface` at all?